File: readability/readability.py
import re
pattern = re.compile('[!.?]+')

#getting user input
text = input("Text:  ")
while (len(text) < 1 or text.isdigit()):
    text =  input("Text: ")


words = 0
letters = 0
sentences = 0

#number of words
for word in text.strip().split():
    words += 1


#no of letters
for i in range(len(text)):
    if text[i].isalpha():
       letters += 1


#no of sentences
for sentence in re.findall(pattern, text.strip()):
    sentences += 1

#conditionals
L = letters /words * 100
S = sentences / words * 100
cl_index = 0.0588 * L - 0.296 * S - 15.8

# L => average number of letters per 100 words in the text
# S => average number of sentences per 100 words in the text

if cl_index > 16:
    print("Grade 16+")
elif cl_index < 1:
    print("Before Grade 1")
else:
    print("Grade %s" %int(round(cl_index)))


print("Letters: %s" %letters)
print("Sentence: %s" %sentences)
print("Words: %s" %words)

File: readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
int main(void)
{
    string text = get_string("Enter Text: ");

    int words = 0;
    int letters = 0;
    int sentence = 0;

    for (int i=0; i < strlen(text); i++)
    {
        if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            sentence+=1;
        }

    }
    // printf("%i",sentence);
    // printf("\n");

    for (int i= 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
    }
    // printf("%i", letters);
    // printf("\n");

    for (int i = 0; i <= strlen(text); i++)
    {
        if (isspace(text[i]) || text[i] == '\0' || text[i] == ' ')
        {
            words += 1;
        }
    }
    // printf("%i", words);
    // printf("\n");

    float L = ((float) letters/words) * 100;
    float S = ((float) sentence/words) * 100;
    float Coleman_Liau_index = 0.0588 * L - 0.296 * S - 15.8;

    if (Coleman_Liau_index >= 16)
    {
        printf("Grade 16+");
    }
    else if (Coleman_Liau_index < 1)
    {
        printf("Before Grade 1");
    }
    else if (Coleman_Liau_index >= 1 || Coleman_Liau_index < 17)
    {
        printf("Grade %i", (int) round(Coleman_Liau_index));
    }
    printf("\n");
    // printf("%d\n%d\n%d\n", words,letters,sentence);

}